src/gui/providers/ogr/qgsgeopackageitemguiprovider.cpp

```diff
@@ -171,6 +173,13 @@ void QgsGeoPackageItemGuiProvider::deleteGpkg()
       else
       {
         QMessageBox::information( nullptr, title, tr( "GeoPackage deleted successfully." ) );
+        // If the deleted file was a stored connection, remove it too
+        if ( ! name.isEmpty() )
+        {
+          QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) };
+          std::unique_ptr<QgsAbstractProviderConnection> conn { md->createConnection( name ) };
+          conn->remove( name );
```

Comment:
Can't we just do

    md->deleteConnection( name );

instead?

Reply:
yeah, cleaner.